core/src/transport.rs

```diff
@@ -129,17 +140,27 @@ pub trait Transport {
     ///
     /// If [`TransportError::MultiaddrNotSupported`] is returned, it may be desirable to
     /// try an alternative [`Transport`], if available.
-    fn dial(&mut self, addr: Multiaddr) -> Result<Self::Dial, TransportError<Self::Error>>;
+    fn dial(
+        &mut self,
+        addr: Multiaddr,
+        opts: DialOpts,
```

Comment:
Wouldnt it be better to just use `DialOpts` since it would contain the address as well?

Reply:
I am open to that, what does @thomaseizinger think?

Reply:
`DialOpts` from `libp2p-swarm` contains a lot more info. For example, it also controls the concurrency factor that the `Transport` itself doesn't know anything about.